### ISSUE TITLE
<mj-link> encodes curly braces in URL, breaking template variables

### ISSUE THREAD

**Issue by joestrouth1** [mjmlio/mjml#664]
As mentioned in #599 , though it's a separate issue.

[Demo on Try It Live](https://mjml.io/try-it-live/S1DwKl91Z)

The curly braces used in common template variable syntax like `{email}` get encoded when used in the the href of an `<mj-link>` tag.

For example:

```html
<mj-inline-links base-url="https://mjml.io">
            <mj-link href="/gettings-started-onboard?utm_source=email&utm_email={email}" color="#ffffff">Getting started</mj-link>
```

becomes

```html
<a href="https://mjml.io/gettings-started-onboard?utm_source=email&utm_email=%7Bemail%7D" style="display:inline-block;text-decoration:none;text-transform:uppercase;color:#ffffff;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;font-weight:normal;line-height:22px;padding:15px 10px;">Getting started</a>
```

Seems odd, since ampersands are not encoded. If there's a method of escaping this please point me in the right direction.

Discovered this in my own work when exporting from the MJML desktop app running v^3.3.2, found it to be the same on Try It Live. It was mentioned in #599 as potentially being related to #601 but that issue has since been resolved without effect. 

Apologies if this has been described or resolved elsewhere, tried to UTFSE and didn't find it specifically called out anywhere.

P.S. - [The Try It Live page for `<mj-navbar>`](https://mjml.io/try-it-live/components/navbar) links to `href="/gettings-started-onboard"` which returns 404.

**Comment by iRyusa** [comment]
Hi @joestrouth1

Looks like every `a` is encoded at some point... I don't know if we can do anything about it yet, in MJML4 this should no longer happen

**Comment by joestrouth1** [comment]
@iRyusa No worries. Looking forward to MJML 4. Just thought it was odd, given that ampersands don't get encoded, despite their more commonly worrisome potential for ambiguity

**Comment by kmcb777** [comment]
Tested on current MJML 4 version, the curly braces were preserved so it seems ok now

**Comment by ngarnier** [comment]
This is now fixed in MJML 4.0.0-beta.2 so I'm closing! 

Please upgrade with `npm install mjml@next` to use the latest beta.
